sdk/plugin/v5/plugin.go

```diff
@@ -0,0 +1,166 @@
+package plugin
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	plugin "github.com/hashicorp/go-plugin"
+	"github.com/hashicorp/vault/sdk/helper/consts"
+	"github.com/hashicorp/vault/sdk/helper/pluginutil"
+	"github.com/hashicorp/vault/sdk/logical"
+	bplugin "github.com/hashicorp/vault/sdk/plugin"
+	"github.com/hashicorp/vault/sdk/plugin/pb"
+)
+
+// BackendPluginClient is a wrapper around backendPluginClient
+// that also contains its plugin.Client instance. It's primarily
+// used to cleanly kill the client on Cleanup()
+type BackendPluginClient struct {
+	client pluginutil.PluginClient
+
+	logical.Backend
+}
+
+type ContextKey string
+
+func (c ContextKey) String() string {
+	return "plugin" + string(c)
+}
+
+const ContextKeyPluginReload = ContextKey("plugin-reload")
+
+// Cleanup cleans up the go-plugin client and the plugin catalog
+func (b *BackendPluginClient) Cleanup(ctx context.Context) {
+	_, ok := ctx.Value(ContextKeyPluginReload).(string)
+	if !ok {
+		b.Backend.Cleanup(ctx)
+		b.client.Close()
+		return
+	}
+	b.Backend.Cleanup(ctx)
+	b.client.Reload()
+}
+
+// NewBackend will return an instance of an RPC-based client implementation of the backend for
+// external plugins, or a concrete implementation of the backend if it is a builtin backend.
+// The backend is returned as a logical.Backend interface. The isMetadataMode param determines whether
```

Comment:
Apologies if my understanding is still rudimentary 😄 Wondering if this comment around the `isMetadataMode` param is still true since this function doesn't seem to accept that param anymore.

I see we're passing AutoMTLS as `true` in the config, guessing `isMetadataMode` isn't needed and AutoMTLS is the default behavior for v5? Thanks for clarifying in advance! 🙏🏼

Reply:
Good catch. I need to update that comment since we are no longer passing that param